### ISSUE TITLE
Self-host web fonts

### ISSUE THREAD

**Issue by lupinia** [lupinia/awi#132]
Prompted by a friend on Mastodon, this is something that's been on the to-do list for a LONG time, and I never opened an issue to actually do it.  And since there's no longer a performance incentive to host common web assets from globally shared CDNs, increasing my web independence just makes sense.

**Comment by lupinia** [comment]
The obnoxious part of doing this is simply downloading all the files and giving them sensible names so I can find and reference them again.  But, the .woff2 files are just files, so I can simply host them somewhere else, and rewrite the CSS file that references them.  Which probably doesn't even need to be its own separate file, but I'll experiment with that and see what happens.

**Comment by lupinia** [comment]
Google Fonts serves [different versions of each file based on user agent](https://developers.google.com/fonts/docs/technical_considerations), and their filenames are complete gibberish, which makes it impossible to tell which ones are duplicates unless you keep their names (and if you do, it then becomes impossible to tell which files contain which fonts).  So I tried downloading all the fonts from their [Github repo](https://github.com/google/fonts), but that appears to only contain the TTF files (which I already have), not the WOFF or WOFF2 files.  I found a converter that *might* work, [google-webfonts-helper](https://github.com/majodev/google-webfonts-helper), which apparently used to be linked from the official Google documentation but isn't anymore, but it still works.  So I think I can use that to download the specific fonts I'm looking for, and hope that they're assembled correctly.  This REALLY should be easier.

**Comment by lupinia** [comment]
This turned into a ridiculous project that consumed my entire day, but I managed to pull it off.  I had to use the aforementioned google-webfonts-helper to download the files themselves, but it was insufficient to build replacement CSS files, and manual copy and paste was prohibitively tedious because there ended up being a thousand different font files.  So I wrote a standalone Python script to build those.  Conclusion:  This is *wildly* impractical, for reasons *entirely* of Google's making, but worth doing for privacy reasons, not to mention the fact that I do *not* trust Google for longevity - my website has outlived almost everything they've ever created, and I expect it to outlive everything they haven't already destroyed on their own.
